### PR TITLE
*: add bazel rules and make bazel compatible

### PR DESCRIPTION
**Pull request by otan**

### CODE REVIEW

File: unix/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        "//conditions:default": [],
+    }),
+    importpath = "",
+    visibility = ["//visibility:public"],
+    deps = ["@com_github_knz_go_libedit//common:go_default_library"],
+)
+
+go_library(
+    name = "unix",
+    cdeps = [":libedit_unix_cdeps"], # TODO: make this work on non-mac.
```
Comment (knz) [inline unix/BUILD.bazel]: Can you explain what doesn't work

Comment (otan) [reply unix/BUILD.bazel]: it needs to do a `select` based on operating system, and compile `unix/src` instead for linux.
also very draft-y at the moment -- just building this atm to make it easier to replace as a go module.

Comment (knz) [reply unix/BUILD.bazel]: Thanks for explaining. Give me a heads up when you need something from me

File: unix/src/c-libedit/linux-build/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,13 @@
+cc_library(
+    name = "linux-build",
+    srcs = glob([
+        "**/*.c",
+        "**/*.h",
+    ]),
+    hdrs = glob([
+        "**/*.h",
+        "**/*.c",
```
Comment (irfansharif) [inline unix/src/c-libedit/linux-build/BUILD.bazel:9]: Did we need to glob over the *.c files for hdrs? And vice versa for *.h files for srcs.

File: unix/src/c-libedit/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,16 @@
+cc_library(
+    name = "c-libedit",
+    srcs = glob([
+        "*.c",
+        "*.h",
+        "editline/readline.h",
+    ]),
+    hdrs = glob([
+        "*.h",
```
Comment (irfansharif) [inline unix/src/c-libedit/BUILD.bazel:9]: Ditto re: globbing over *.{h,c}

File: unix/BUILD.bazel [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        "wrap-vi.c",
+        "wrap-vis.c",
+        "wrap-wcsdup.c",
+    ],
+    cdeps = [
+        ":libedit_unix_cdeps",
+    ],
+    cgo = True,
+    clinkopts = select({
+        "@io_bazel_rules_go//go/platform:android": [
```
Comment (irfansharif) [inline unix/BUILD.bazel:99]: Can we skip ios/android builds? I'm not sure why bazel inserts them. The go package doesn't refer to these explicitly. Maybe not worth doing, I'm just curious how they got auto-generated.

File: unix/BUILD.bazel [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+            "//unix/src/c-libedit:c-libedit",
+        ],
+        "//conditions:default": [],
+    }),
+)
+
+cc_library(
+    name = "libedit_cdeps",
+    srcs = glob([
+        "shim/**/*.c",
```
Comment (irfansharif) [inline unix/BUILD.bazel:47]: Ditto.

**Review by irfansharif — APPROVED** [review]
LGTM. Perhaps worth adding a few words to the README about how to build the target, and few words in commit message about how the autogenerated BUILD files were created (+ the deltas added on top).

File: unix/src/c-libedit/BUILD.bazel (hunk 3 above)
Comment (otan) [reply unix/src/c-libedit/BUILD.bazel:9]: yes, because there are a lot of #include "x.c"s in here.

File: unix/BUILD.bazel (hunk 4 above)
Comment (otan) [reply unix/BUILD.bazel:99]: i can't figure out how to *ungenerate* them, happens in the CRDB repo too.

**Comment by otan** [comment]
https://github.com/otan-cockroach/go-libedit/tree/gazelle_it_2 is the solution that works, closing this one for now, firm commit later!

**Pull request closed without merging**